Confuser.Runtime/IntegrityProtection.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Confuser.Runtime
+{
+    internal static class IntegrityProtection
+    {
+        internal static void Initialize()
+        {
+            var bas = new StreamReader(typeof(IntegrityProtection).Assembly.Location).BaseStream;
+            var file = new BinaryReader(bas);
+            var file2 = File.ReadAllBytes(typeof(IntegrityProtection).Assembly.Location);
+
+            var byt = file.ReadBytes(file2.Length - 32);
+            var a = Hash(byt);
+            file.BaseStream.Position = file.BaseStream.Length - 32;
+            string b = Encoding.ASCII.GetString(file.ReadBytes(32));
+
+            if (a != b)
+                throw new BadImageFormatException();
+        }
+
+        internal static string Hash(byte[] metin)
+        {
+            MD5CryptoServiceProvider md5 = new MD5CryptoServiceProvider();

```

Comment:
MD5 is extremely obsolete, even something as old as SHA-1 is faster and more secure.